source/common/buffer/zero_copy_input_stream_impl.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "common/buffer/buffer_impl.h"
+
+#include "google/protobuf/io/zero_copy_stream.h"
+
+namespace Envoy {
+
+namespace Buffer {
+
+class ZeroCopyInputStreamImpl : public virtual google::protobuf::io::ZeroCopyInputStream {
+public:
+  // Create input stream with one buffer, and finish immediately
```

Comment:
I'm not sure if there is any easy way to do this, but in this case, it would be nice if we didn't have to allocate a new buffer on the stack, just to move into, and then operate. IMO we should be able to operate directly on the input buffer since it's non-const anyway. I would just put in a TODO here for a potential perf optimization.